flixel/input/gamepad/ButtonIndex.hx

```diff
@@ -0,0 +1,405 @@
+package flixel.input.gamepad;
+import flixel.input.gamepad.ButtonID;
+import flixel.input.gamepad.FlxGamepad.FlxGamepadAnalogStick;
+import flixel.input.gamepad.FlxGamepad.GamepadModel;
+
+/**
+ * ...
+ * @author larsiusprime
+ */
+class ButtonIndex
+{
+	public var model:GamepadModel;
+	
+	public function new(Model:GamepadModel) 
+	{
+		model = Model;
+	}
+	
+	/**
+	 * Given a ButtonID, return the raw hardware code
+	 * @param	buttonID the "universal" ButtonID
+	 * @return	the raw hardware code
+	 */
+	
+	public function getRaw(buttonID:ButtonID):Int
+	{
+		return switch(model)
+		{
+			case Logitech: getRawLogitech(buttonID);
+			case OUYA: getRawOUYA(buttonID);
+			case PS3: getRawPS3(buttonID);
+			case PS4: getRawPS4(buttonID);
+			case Xbox: getRawXbox(buttonID);
+			case XInput: getRawXInput(buttonID);
+			default: -1;
+		}
+	}
+	
+	/**
+	 * Given a raw hardware code, return the "universal" ButtonID
+	 * @param	RawID	the raw hardware code
+	 * @return	the "universal" ButtonID
+	 */
+	
+	public function getBtn(RawID:Int):ButtonID
+	{
+		return switch(model)
+		{
+			case Logitech: getBtnLogitech(RawID);
+			case OUYA: getBtnOUYA(RawID);
+			case PS3: getBtnPS3(RawID);
+			case PS4: getBtnPS4(RawID);
+			case Xbox: getBtnXbox(RawID);
+			case XInput: getBtnXInput(RawID);
+			default: NONE;
+		}
+	}
+	
+	/**
+	 * Given a ButtonID, return the raw AnalogStick axes data structure
+	 * @param	buttonID	the "universal" ButtonID
+	 * @return	structure containing raw analog stick axes integer codes
+	 */
+	
+	public function getRawAnalogStick(buttonID:ButtonID):FlxGamepadAnalogStick
+	{
+		if (buttonID == ButtonID.LEFT_ANALOG_STICK)
+		{
+			return switch(model)
+			{
+				case Logitech: LogitechButtonID.LEFT_ANALOG_STICK;
+				case OUYA: OUYAButtonID.LEFT_ANALOG_STICK;
+				case PS3: PS3ButtonID.LEFT_ANALOG_STICK;
+				case PS4: PS4ButtonID.LEFT_ANALOG_STICK;
+				case Xbox: XboxButtonID.LEFT_ANALOG_STICK;
+				case XInput: XInputButtonID.LEFT_ANALOG_STICK;
+				default: null;
+			}
+		}
+		if (buttonID == ButtonID.RIGHT_ANALOG_STICK)
+		{
+			return switch(model)
+			{
+				case Logitech: LogitechButtonID.RIGHT_ANALOG_STICK;
+				case OUYA: OUYAButtonID.RIGHT_ANALOG_STICK;
+				case PS3: PS3ButtonID.RIGHT_ANALOG_STICK;
+				case PS4: PS4ButtonID.RIGHT_ANALOG_STICK;
+				case Xbox: XboxButtonID.RIGHT_ANALOG_STICK;
+				case XInput: XInputButtonID.RIGHT_ANALOG_STICK;
+				default: null;
+			}
+		}
+		return null;
+	}
+	
+	/*****************************************/
+	
+	public inline function getRawOUYA(buttonID:ButtonID):Int
```

Comment:
They're not commented out, I think it's getting confused because of the /****/ separator